script/libtest.sh

```diff
@@ -12,37 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set -e
+
 export TEST_MAX_WAIT_SEC=300
 
 ## k8s specific Helper functions
-k8s_wait_for_pod() {
-    namespace=${1:?}
-    labelSelector=${2:?}
-    condition=${3:?}
-    echo "Waiting for pod '${@:2}' to be ${condition} ... "
-    local -i cnt=${TEST_MAX_WAIT_SEC:?}
-
-    # Retries just in case it is not stable
-    local -i successCount=0
-    while [ "$successCount" -lt "3" ]; do
-        if kubectl get pod -a -n "$namespace" -l "$labelSelector" | grep -q "$condition"; then
-            ((successCount=successCount+1))
-        fi
-        ((cnt=cnt-1)) || return 1
-        sleep 1
-    done
-
-}
-k8s_wait_for_pod_ready() {
-    namespace=${1:?}
-    labelSelector=${2:?}
-    k8s_wait_for_pod $namespace $labelSelector Running
-}
-
-k8s_wait_for_pod_completed() {
+k8s_wait_for_deployment() {
     namespace=${1:?}
-    labelSelector=${2:?}
-    k8s_wait_for_pod $namespace $labelSelector Completed
+    deployment=${2:?}
+    echo "Waiting for deployment ${deployment} to be successfully rolled out"
+    kubectl rollout status --namespace $namespace deployment ${deployment}
```

Comment:
Without storing the exit code, isn't this going to return right away here because of the `set -e`? Have you tried this code locally? This is the way I tested mine https://github.com/kubeapps/kubeapps/pull/833#issuecomment-439526636

Reply:
Yes, it will. In fact the rest of the function code (printing the exit code) will only happen if the exit code is 0. I can "disable" errors to print the error code in any case but this function should fail if the deployment fails to rollout.